### DISCUSSION TITLE
Can REFCLK_x_PLL_NW frequency be Decimal?

### DISCUSSION THREAD

**Discussion by henryding-emcore** [polarfire-soc/polarfire-soc-documentation#120] [Bare metal embedded software] [answered]
**I got following compilation errors from Softconsole if I use REFCLK_x_PLL_NW as the MSS PLL reference clock source and RTC clock source and set PLL_NW frequency to be a decimal, i.e. 122.88 MHz in MSS Configurator**

...\src\boards\icicle-kit-es/fpga_design_config/clocks/hw_mss_clks.h:27: error: unterminated #ifndef
 #ifndef HW_MSS_CLKS_H_ 

**The reason for causing this error is:**
python interpreter running is version 3
  File "../src/platform/soc_config_generator/mpfs_configuration_generator.py", line 711, in <module>
  File "../src/platform/soc_config_generator/mpfs_configuration_generator.py", line 707, in main_config_generator
  File "../src/platform/soc_config_generator/mpfs_configuration_generator.py", line 529, in generate_header_files
  File "../src/platform/soc_config_generator/mpfs_configuration_generator.py", line 422, in generate_header
  File "../src/platform/soc_config_generator/mpfs_configuration_generator.py", line 338, in generate_register
ValueError: invalid literal for int() with base 16: '122880000.0'
make[1]: [makefile:91: pre-build] Error 1 (ignored)

**There is no issue if I give an integer number, like 122 MHz.** 

**Thanks in advance!**

**Comment by hughbreslin** [comment, marked as answer]
Hi @henryding-emcore 

The values you're using sound valid :) 

The issue here is in the XML generation from the looks of it - I recreated your set up below (value of 122.88MHz left and 125MHz right):
![image](https://user-images.githubusercontent.com/44862881/177808375-ba7c2bfc-d421-4147-85b4-a2e09263627d.png)


The MHz value is converted to Hz in the XML and when you use a decimal value for the clock source in MHz the resulting XML has a ".0" appended to the Hz value. 

I was able to successfully generate the config files by removing the ".0" at the end of the `MSS_EXT_SGMII_REF_CLK` field. If you search your XML for `MSS_EXT_SGMII_REF_CLK` you should see the ".0", can you confirm removing this resolves the build error for you?

**Reply by henryding-emcore** [reply]
Yes, your solution resolves the problem. Thanks so much, Hugh!

**Reply by hughbreslin** [reply]
Great 🙂 We'll get this fixed for a future release of the configurator
